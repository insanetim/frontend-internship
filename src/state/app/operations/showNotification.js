import { createLogic } from 'redux-logic'
import { notification } from 'antd'
import { path } from 'ramda'

import { NOTIFICATION_DURATION } from 'src/constants'
import * as types from '../types'

const showNotification = createLogic({
  type: types.SHOW_NOTIFICATION,
  latest: true,

  process({ action }) {
    const type = path(['payload', 'type'], action)
    const message = path(['payload', 'message'], action)

    notification[type]({ message, duration: NOTIFICATION_DURATION })
  }
})

export default showNotification
