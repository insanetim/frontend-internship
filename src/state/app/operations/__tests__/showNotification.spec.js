import { notification } from 'antd'

import showNotification from '../showNotification'

describe('showNotification', () => {
  it('has valid attributes', () => {
    expect(showNotification).toMatchSnapshot()
  })

  it('calls notification success', () => {
    notification.success = jest.fn()
    showNotification.process(
      {
        action: {
          payload: { type: 'success' }
        }
      },
      jest.fn(),
      jest.fn()
    )
    expect(notification.success).toHaveBeenCalled()
  })
})
