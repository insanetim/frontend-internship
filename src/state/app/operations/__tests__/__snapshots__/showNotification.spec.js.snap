// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`showNotification has valid attributes 1`] = `
{
  "cancelType": undefined,
  "debounce": 0,
  "latest": true,
  "name": undefined,
  "process": [Function],
  "processOptions": {
    "dispatchMultiple": true,
  },
  "throttle": 0,
  "transform": undefined,
  "type": "app/SHOW_NOTIFICATION",
  "validate": undefined,
  "warnTimeout": 60000,
}
`;
